internal/cmd/cleanenv/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+func main() {
+	app := App{
+		Args:   os.Args[1:],
+		Env:    os.Environ(),
+		Out:    os.Stdout,
+		ErrOut: os.Stderr,
+	}
+	err := app.Run()
+	if err != nil {
+		fmt.Fprintln(app.ErrOut, err)
+		exitCode := 1
+		var exitErr *exec.ExitError
+		if errors.As(err, &exitErr) {
+			exitCode = exitErr.ExitCode()
+		}
+		os.Exit(exitCode)
+	}
+}
+
+type App struct {
```

Comment:
I designed this with extensibility and testability in mind, hence extra bits like this. It should help if we later decide to share the tool more widely.